File: Dating-App/lib/helpers/chat_quality_helper.dart
import 'package:dating_app/datas/user.dart';

class ChatQualityHelper {
  static List<String> getConversationStarters(User otherUser) {
    List<String> starters = [];

    if (otherUser.userDegree.isNotEmpty) {
      starters.add("What inspired you to pursue a ${otherUser.userDegree}?");
    }

    if (otherUser.userOccupation.isNotEmpty) {
      starters.add("How's your experience been working as a ${otherUser.userOccupation}?");
    }

    if (otherUser.userResearchInterests.isNotEmpty) {
      starters.add("I saw you're interested in ${otherUser.userResearchInterests.first}. What's the most exciting development in that field recently?");
    }

    if (otherUser.userFutureGoals.isNotEmpty) {
      starters.add("Your goal of ${otherUser.userFutureGoals.first} is impressive. How are you planning to achieve it?");
    }

    // Generic fallback if user profile is sparse
    if (starters.isEmpty) {
      starters.add("Hello! I'd love to learn more about your academic journey.");
      starters.add("Hi! What's the most interesting thing you're working on right now?");
    }

    return starters;
  }

  static List<String> getIceBreakers() {
    return [
      "If you could have a conversation with any famous scholar, who would it be?",
      "What's one piece of advice you'd give to someone just starting in your field?",
      "Early bird or night owl for study/work sessions?",
      "What's your go-to productivity hack?",
      "If you weren't in your current profession/degree, what would you be doing?"
    ];
  }
}

File: React/Dating-App/src/utils/chatQualityHelper.ts
import type { User } from '../types/user';

export class ChatQualityHelper {
  static getConversationStarters(otherUser: User): string[] {
    const starters: string[] = [];

    if (otherUser.user_degree) {
      starters.push(`How is your ${otherUser.user_degree} journey going?`);
    }

    if (otherUser.user_institution) {
      starters.push(`What is the most interesting thing about ${otherUser.user_institution}?`);
    }

    if (otherUser.user_research_interests && otherUser.user_research_interests.length > 0) {
      starters.push(`I noticed you're interested in ${otherUser.user_research_interests[0]}. Tell me more!`);
    }

    if (otherUser.user_occupation) {
      starters.push(`What inspired you to specialize in ${otherUser.user_occupation}?`);
    }

    if (otherUser.user_future_goals && otherUser.user_future_goals.length > 0) {
      starters.push(`Your goal of ${otherUser.user_future_goals[0]} is very ambitious. How are you working towards it?`);
    }

    return starters.length > 0 ? starters : ["Hi! Tell me about your academic journey."];
  }

  static getIceBreakers(): string[] {
    return [
      "If you could have dinner with any scientist/expert in your field, who would it be?",
      "What is the most challenging research paper you've ever read?",
      "Late-night study sessions: library or coffee shop?",
      "What is your 'holy grail' achievement in your profession?",
      "Digital notes or physical notebooks for your research?"
    ];
  }
}
